js/pivx_shielding.js

```diff
@@ -88,11 +94,36 @@ export default class PIVXShielding {
   }
 
   /**
-   * Creates a transaction, sending `amount` satoshis to the addresses
-   * @param {{address: String, amount: String}[]} targets
+   * Creates a transaction, sending `amount` satoshis to the address
+   * @param {{address: String, amount: Number}} target
+   */
+  createTransaction({address, amount, blockHeight}) {
+    const { txid, txhex, nullifiers } = this.shieldMan.create_transaction(
+      this.unspentNotes,
+      this.extsk,
+      address,
+      this.getNewAddress(),
+      amount,
+      blockHeight,
+      this.isTestnet
+    );
+
+    this.pendingTransactions.set(txid, nullifiers);
+
+    return txhex;
+  }
+
+  /**
+   * Signals the class that a transaction was sent successfully
+   * and the notes can be marked as spent
+   * @param{String} txid - Transaction id
    */
-  createTransaction(targets) {
-    throw new Error("Not implemented");
+  finalizeTransaction(txid) {
```

Comment:
Can you also add a function that remove an element of pendingTransactions without removing the corresponding note in unspentNotes? Can be useful when the tx fails on MPW